.github/workflows/pr-generate-icons.yml

```diff
@@ -0,0 +1,40 @@
+name: Generate icon components
```

Comment:
Haven't tested this and the publish pipelines yet, but this is the general idea of how they'd work.